engine/proofview.ml

```diff
@@ -373,32 +373,26 @@ let tclTRYFOCUS i j t = tclFOCUS ~nosuchgoal:(tclUNIT ()) i j t
 let tclFOCUSLIST ?(nosuchgoal=tclZERO (NoSuchGoals 0)) l t =
   let open Proof in
   Comb.get >>= fun comb ->
-    let n = CList.length comb in
-    (* First, remove empty intervals, and bound the intervals to the number
-       of goals. *)
-    let sanitize (i, j) =
-      if i > j then None
-      else if i > n then None
-      else if j < 1 then None
-      else Some ((max i 1), (min j n))
-    in
-    let l = CList.map_filter sanitize l in
+  let n = CList.length comb in
+  (* First, remove empty intervals, and bound the intervals to the number
+     of goals. *)
+  let ok (i, j) = 1 <= i && i <= j && j <= n in
+  if not (CList.for_all ok l) then tclZERO (NoSuchGoals 0)
```

Comment:
The comment above becomes inaccurate since we don't remove empty intervals, we raise an error if there are some.
Furthermore, it seems a bit strange to decide to raise a NoSuchGoal error if the problem is a nonsensical interval.
Finally, why use `tclZERO (NoSuchGoals 0)` here instead of `nosuchgoal`?

Reply:
I am fine with the `NoSuchGoal` error for nonsensical intervals, but indeed maybe using the generalized error is probably better.

Reply:
NoSuchGoal is there only because this is rebased from when I first wrote it, which was before nosuchgoal

Reply:
>Furthermore, it seems a bit strange to decide to raise a NoSuchGoal error if the problem is a nonsensical interval.

nosuchgoal could be called "invalid selector" instead I guess
I don't care enough to make a PR for it though